api/pubsub/pubsub.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package pubsub implements the API for streaming pubsub messages between api
+// servers.
+package pubsub
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/apiserver/params"
+)
+
+// MessageWriter is the interface that allows sending pub/sub messges to the
+// server.
+type MessageWriter interface {
+	// ForwardMessage forwards the given message to the server.
+	ForwardMessage(*params.PubSubMessage) error
+	Close() error
+}
+
+// API provides access to the pubsub API.
+type API struct {
+	connector base.StreamConnector
+}
+
+// NewAPI creates a new client-side pubsub API.
+func NewAPI(connector base.StreamConnector) *API {
+	return &API{connector: connector}
+}
+
+// MessageWriter returns a new message writer interface value
+// which must be closed when finished with.
+func (api *API) MessageWriter() (MessageWriter, error) {
```

Comment:
I would like to have an "Open" in the name of this method so it "hints" that its something that must be closed, people might forget to otherwise.